velox/expression/tests/ExpressionFuzzer.h

```diff
@@ -337,41 +318,60 @@ class ExpressionFuzzer {
   // --assign_function_tickets startup flag .
   std::unordered_map<std::string, int> functionsToTickets_;
 
-  /// The remaining levels of expression nesting. It's initialized by
-  /// FLAGS_max_level_of_nesting and updated in generateExpression(). When its
-  /// value decreases to 0, we don't generate subexpressions anymore.
-  int32_t remainingLevelOfNesting_;
-
   /// We allow the arg generation routine to be specialized for particular
   /// functions. This map stores the mapping between function name and the
   /// overridden method.
   using ArgsOverrideFunc = std::function<std::vector<core::TypedExprPtr>(
       const CallableSignature& input)>;
+
   std::unordered_map<std::string, ArgsOverrideFunc> funcArgOverrides_;
 
-  std::shared_ptr<core::QueryCtx> queryCtx_{std::make_shared<core::QueryCtx>()};
-  std::shared_ptr<memory::MemoryPool> pool_{memory::addDefaultLeafMemoryPool()};
-  core::ExecCtx execCtx_{pool_.get(), queryCtx_.get()};
-  test::ExpressionVerifier verifier_;
+  // This pool is needed to generate vectors when constant expressions get
+  // generated.
+  std::shared_ptr<memory::MemoryPool> pool_;
+
+  std::shared_ptr<VectorFuzzer> vectorFuzzer_;
```

Comment:
I thought having `VectorFuzzer& vectorFuzzer_` would be simpler. Is there a reason why you prefer std::shared_ptr<VectorFuzzer>?